Add feature flag to disable core profiler bundle for select users

chihsuan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes #56080.

This PR removes the fallback system for core-provided recommendations in the core profiler to ensure partnerships changes can be handled efficiently. The implementation includes:

- Feature flag `disable-core-profiler-fallback` for controlling the removal. The feature is enabled by default. It's added in case it needs to be easily reverted.
- 50% gradual rollout using `woocommerce_remote_variant_assignment`
- Maintained core strings for WooCommerce.com translations
- Add unit tests for all scenarios

(For Bug Fixes) Bug introduced in PR # .


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site
2. Go to `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`
3. Disable `Display suggestions within WooCommerce` option
4.  Set `woocommerce_remote_variant_assignment` to 60
5. Go to Core Profiler and continue to `Tell us a bit about your store` step
6. Click on the Continue button
7. The free features plugin step should be skipped
8. Change the `woocommerce_remote_variant_assignment` to 61
9. Repeat the steps above and verify the core profiler plugin step is shown

**Regression test: It should not skip when:**
- User is eligible for rollout
- Marketplace suggestion enabled

**WCCOM API error test: It should skip when:**
- User is eligible for rollout
- Marketplace suggestions enabled
- WCCOM error set in Force errors on woocommerce.com requests in WCA Test Helper > Tools  and transient/cache cleared


<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

chihsuan committed:
Add feature flag to disable core profiler bundle for select users

This commit introduces a new feature flag 'disable-core-profiler-fallback' that allows for selective removal of the core profiler bundle. When enabled, the bundle is removed for 50% of users based on a remote variant assignment. The changes include:

- Update configuration files to include the new feature flag
- Modify DefaultFreeExtensions to conditionally remove the core profiler bundle
- Add comprehensive test cases to verify the feature flag's behavior

chihsuan committed:
Add changelog

ilyasfoo reviewed (dismissed):
Edit: I misread! Nevermind my comment

adrianduffell reviewed (approved):
Thanks @chihsuan , LGTM 🚀 

Tested with JN per testing instructions. Video confirming free features screen fallback removed when suggestion disabled.

https://github.com/user-attachments/assets/46ff4326-f84d-4eed-a514-e265e7bba394

ilyasfoo commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -95,6 +96,21 @@ public static function get_all() {
 		);
 
+		if ( Features::is_enabled( 'disable-core-profiler-fallback' ) ) {
+			$remote_variant_assignment = (int) get_option( 'woocommerce_remote_variant_assignment', 1 ); // 1-120
```
Nit: Should we default to `61`, i.e: to not removing the defaults fallback? In my mind the default behaviour should be the existing behaviour, and feels safer this way

chihsuan replied:
Sounds good! I've made the change in 079671527f5879ad9c1ca735ab09a313b278ffa1. Thanks for the suggestion.

ilyasfoo reviewed (approved):
Thanks for the quick work here, @chihsuan! This tests well. I have a few comments but this looks good to go! 🚢 

### Testing instructions

I think it's good to also to add the following tests, which tests well in my case:
1. Regression test: It should not skip when:
    - User is eligible for rollout 
    - Marketplace suggestion enabled
3. WCCOM API error test: It should skip when:
    - User is eligible for rollout
    - Marketplace suggestions enabled
    - WCCOM error set in `Force errors on woocommerce.com requests` in `WCA Test Helper > Tools ` and transient/cache cleared

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ✅ All good |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Documentation  | Not applicable |
| PR description  | ✅ All good |
| Testing instructions  | Commented |
| Manual testing  | ✅ Works as expected |

#### Scenarios tested

| Scenario  | Result |
| ------------- | ------------- |
| Extensions step skipped when marketplace disabled and option <= 60  | ✅ Functionality worked correctly | 
| Extensions step skipped when marketplace enabled and option <= 60 and WCCOM API error | ✅ Functionality worked correctly |
| Extensions step not skipped when marketplace disabled and option > 60  | ✅ Functionality worked correctly |
| Extensions step not skipped when marketplace enabled and option <= 60  | ✅ Functionality worked correctly |

chihsuan committed:
Adjust default remote variant assignment for core profiler bundle

Update the default value for 'woocommerce_remote_variant_assignment' from 1 to 61, which changes the rollout percentage for the core profiler bundle feature flag

chihsuan committed:
Merge branch 'trunk' into update/disable-core-profiler-free-feature-fallback
